src/expr.h

```diff
@@ -83,6 +83,11 @@ class Expr : public ASTNode {
     virtual Expr *Instantiate(TemplateInstantiation &templInst) const = 0;
 
     virtual bool HasAmbiguousVariability(std::vector<const Expr *> &warn) const;
+
+    /** Returns a string that represents the mangled type and value (for use in
+    mangling of function symbol names). This method is needed when expression is part of a function template
+    instantiation (like ConstExpr or SymbolExpr. */
```

Comment:
Function template parameters is the only case when expressions need to be mangled. Would be good to specifically mention that.